File: in_Python/1833A_musical_puzzle.py
t = int(input())
for _ in range(t):
    num = int(input())
    arr = str(input())
    
    left = 0
    right = 1
    unique = set()
    for i in range(num-1):
        temp = arr[left] + arr[right]
        unique.add(temp)
        left+=1
        right+=1
    print(len(unique))       

File: in_C++/1833A_musical_puzzle.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int t;
    cin >> t;
    while(t--) {
        int num;
        cin >> num;
        string arr;
        cin >> arr;
        int lindex = 0, rindex = 1;
        set<string> unique;
        for (int i = 0;i < num-1;i++){
            string temp = string(1, arr[lindex]) + string(1, arr[rindex]);
            unique.insert(temp);
            lindex++;
            rindex++;
        }
        cout << unique.size() << endl;
    }
    return 0;
}